### PR TITLE
Add multiple track hits together

### PR DESCRIPTION
**Pull request by sigmanotation**
Points from all tracks are now recorded using the same dictionary and converted into an array after the fact. This solves the edge case of recording multiple points for multiple tracks hitting the same pad at the same time. Also fixed some documentation stuff. From some very preliminary testing, there seems to be a speed up of around 25% using one dictionary and then converting only once.

### CODE REVIEW

File: src/attpc_engine/detector/simulator.py [hunk 1]
```diff
@@ -254,7 +293,9 @@ def generate_electrons(
         )
         return electrons
 
-    def generate_point_cloud(self, config: Config, rng: Generator) -> np.ndarray:
+    def generate_point_cloud(
```
Comment (gwm17) [inline src/attpc_engine/detector/simulator.py:296]: Not clear what is happening here. We give it a dictionary to fill and then you return the same dictionary? Should just accept the dict, fill it.

File: src/attpc_engine/detector/simulator.py [hunk 2]
```diff
@@ -299,11 +344,9 @@ def generate_point_cloud(self, config: Config, rng: Generator) -> np.ndarray:
             config.drift_velocity,
             track,
             electrons,
+            points,
         )
 
-        # Wiggle point TBs over interval [0.0, 1.0). This simulates effect of converting
-        # the (in principle) int TBs to floats.
-        points[:, 1] += rng.uniform(low=0.0, high=1.0, size=len(points))
         return points
```
Comment (gwm17) [inline src/attpc_engine/detector/simulator.py:350]: Remove this return

File: src/attpc_engine/detector/simulator.py (hunk 1 above)
Comment (gwm17) [reply src/attpc_engine/detector/simulator.py:296]: Also need type hint for points. Should be 
```python
points: numba.typed.Dict[int, int]
```

File: src/attpc_engine/detector/transporter.py [hunk 3]
```diff
@@ -324,15 +324,16 @@ def transport_track(
...
     electrons: np.ndarray
         1xN array of electrons created each time step (point) of the trajectory.
+    points: numba.typed.Dict
+        A dictionary mapping a unique pad,tb key to the number of electrons.
 
     Returns
     -------
-    numpy.ndarray
-        An Nx3 array representing the simplified point cloud
+    points: numba.typed.Dict[int, int]
```
Comment (gwm17) [inline src/attpc_engine/detector/transporter.py:332]: Same as above, I don't think we should return the dictionary. We fill it in each function, so no need to return a copy (its the same dictionary).

File: src/attpc_engine/detector/transporter.py [hunk 4]
```diff
@@ -342,12 +343,4 @@ def transport_track(
...
-    # Convert to numpy array of [pad, tb, e], now combined over pad/tb combos
-    point_array = np.empty((len(points), 3), dtype=float)
-    for idx, point in enumerate(points.items()):
-        tb, pad = unpair(point[0])
-        point_array[idx, 0] = pad
-        point_array[idx, 1] = tb
-        point_array[idx, 2] = point[1]
-
-    return point_array
+    return points
```
Comment (gwm17) [inline src/attpc_engine/detector/transporter.py:346]: Remove this return.

File: src/attpc_engine/detector/simulator.py (hunk 2 above)
Comment (sigmanotation) [reply src/attpc_engine/detector/simulator.py:350]: Done

File: src/attpc_engine/detector/transporter.py (hunk 4 above)
Comment (sigmanotation) [reply src/attpc_engine/detector/transporter.py:346]: Done

**Review by gwm17 — APPROVED** [review]
g2g whenever you finish

**Comment by gwm17** [comment]
For documentation purposes, this continues addressing #33

**Pull request merged**